src/moonlink/src/storage/iceberg/rest_catalog.rs

```diff
@@ -0,0 +1,272 @@
+use async_trait::async_trait;
+use iceberg::table::Table;
+use iceberg::CatalogBuilder;
+use iceberg::Result as IcebergResult;
+use iceberg::{Catalog, Namespace, NamespaceIdent, TableCommit, TableCreation, TableIdent};
+use iceberg_catalog_rest::{
+    RestCatalog as IcebergRestCatalog, RestCatalogBuilder as IcebergRestCatalogBuilder,
+};
+use std::collections::HashMap;
+
+#[derive(Debug)]
+pub struct RestCatalog {
+    catalog: IcebergRestCatalog,
+}
+
+impl RestCatalog {
+    #[allow(dead_code)]
+    pub async fn new(
+        builder: IcebergRestCatalogBuilder,
```

Comment:
curious why do we pass catalog builder from caller, instead of constructing it internally?
Are you planning to do dependency injection for testing?